jlm/rvsdg/gamma.hpp

```diff
@@ -465,6 +465,33 @@ inline gamma_node::gamma_node(jlm::rvsdg::output * predicate, size_t nalternativ
       new gamma_input(this, predicate, ctltype::Create(nalternatives))));
 }
 
+/**
+ * Represents a region argument in a gamma subregion.
+ */
+class GammaArgument final : public argument
+{
+  friend gamma_node;
+
+public:
+  ~GammaArgument() noexcept override;
+
+private:
+  GammaArgument(
+      rvsdg::region & region,
+      structural_input & input,
+      std::shared_ptr<const rvsdg::type> type)
+      : argument(&region, &input, std::move(type))
+  {}
+
+  static GammaArgument &
+  Create(rvsdg::region & region, structural_input & input, std::shared_ptr<const rvsdg::type> type)
```

Comment:
Does the create method need a type specified, or can it get its type from the `input`?

Reply:
Remove the type argument.